<!--

  turtlebot3_imuodompt2.bag links:
	lidar link: camera_depth_optical_frame
		depth is z-axis of camera_depth_optical_frame
		no rot from base_link, 
	imu link: base_link

	
	12/9/21 imu_link_joint was 180 deg rot x-axis, might need to change to 
	no rot for turtlebot3 bag files.
-->

<robot name="cartographer_backpack_3d">
  <material name="orange">
    <color rgba="1.0 0.5 0.2 1" />
  </material>
  <material name="gray">
    <color rgba="0.2 0.2 0.2 1" />
  </material>

  <link name="imu_link">
    <visual>
      <origin xyz="0.0 0.0 0.0" />
      <geometry>
        <box size="0.06 0.04 0.02" />
      </geometry>
      <material name="orange" />
    </visual>
  </link>

  <link name="camera_depth_optical_frame">
    <visual>
      <origin xyz="0.0 0.0 0.0" />
      <geometry>
        <cylinder length="0.07" radius="0.05" />
      </geometry>
      <material name="gray" />
    </visual>
  </link>

  <link name="vertical_vlp16_link">
    <visual>
      <origin xyz="0.0 0.0 0.0" />
      <geometry>
        <cylinder length="0.07" radius="0.05" />
      </geometry>
      <material name="gray" />
    </visual>
  </link>

  <link name="base_link" />

  <joint name="imu_link_joint" type="fixed">
    <parent link="base_link" />
    <child link="imu_link" />
    <origin xyz="0 0 0" rpy="0 0 0" />
  </joint>

  <joint name="camera_depth_optical_frame_joint" type="fixed">
    <parent link="base_link" />
    <child link="camera_depth_optical_frame" />
    <origin xyz="0.01 0. 0.19" rpy="0. 0. 0." />
  </joint>

  <joint name="vertical_vlp16_link_joint" type="fixed">
    <parent link="base_link" />
    <child link="vertical_vlp16_link" />
    <origin xyz="0.19 0. 0.04" rpy="0. 1.3963 0." />
  </joint>
</robot>
